contracts/RollupChain.sol

```diff
@@ -249,7 +221,10 @@ contract RollupChain is Ownable, Pausable {
             } else if (tnType == tn.TN_TYPE_DEPOSIT_REWARD) {
                 // Update the pending deposit record.
                 dt.DepositRewardTransition memory dp = tn.decodeDepositRewardTransition(_transitions[i]);
-                _checkPendingDeposit(address(0), dp.assetId, dp.amount, _blockId);
+                priorityOperations.checkPendingDeposit(address(0), dp.assetId, dp.amount, _blockId);
+            } else if (tnType == tn.TN_TYPE_UPDATE_EPOCH) {
+                dt.UpdateEpochTransition memory ep = tn.decodeUpdateEpochTransition(_transitions[i]);
```

Comment:
Can we move more all decodings into `PriorityOperations`?

Reply:
done

Reply:
I don't see much benefit in doing so. The current approach minimizes function arg size. Deposit checks have to be done here anyway as there are two related tns

Reply:
Ack. Was trying to further reduce bytecode size of RollupChain.